cli/src/command.rs

```diff
@@ -94,42 +92,31 @@ impl SubstrateCli for Cli {
 			"darwinia" => Box::new(chain_spec::darwinia_config()?),
 			"darwinia-dev" | "dev" => Box::new(chain_spec::darwinia_development_config()),
 			"darwinia-genesis" => Box::new(chain_spec::darwinia_build_spec_config()),
-			path if self.run.force_crab => Box::new(CrabChainSpec::from_json_file(
-				std::path::PathBuf::from(path),
-			)?),
-			path => Box::new(DarwiniaChainSpec::from_json_file(
-				std::path::PathBuf::from(path),
-			)?),
+			path => {
+				let path = PathBuf::from(path);
+				let chain_spec = Box::new(DarwiniaChainSpec::from_json_file(path.clone())?)
+					as Box<dyn ChainSpec>;
+
+				if self.run.force_crab || chain_spec.is_crab() {
```

Comment:
Do we need `chain_spec.is_crab()` here? Is `crab-dev` spec `is_crab` true?

Reply:
Yes. CrabChainSpec shares with all Crab runtimes.

Reply:
crab-dev should not use json file, right?

Reply:
Nope, maybe you want to make a testnet on different machines that you will need to export the chain state to a json file.